aten/src/TH/THTensor.cpp

```diff
@@ -48,14 +48,13 @@ void THTensor_setStorageNd(THTensor *self, THStorage *storage, ptrdiff_t storage
     if (!THTensor_getStoragePtr(self)) {
       THError("Tensor: invalid null storage");
     }
-    auto data_type = THTensor_getStoragePtr(self)->dtype();
     if(storage)
     {
       c10::raw::intrusive_ptr::incref(storage);
       THTensor_stealAndSetStoragePtr(self, storage);
```

Comment:
Probably a pointless question, but - if this is a real steal, like a move, does the ref count get decremented back down in the call? Or is this just sharing the storage?

Reply:
which refcount?

There are two storages here, the tensor's old storage and the tensor's new storage.

For the old storage the refcount is decreased because the tensor no longer points to the old storage at the end/
For the new storage the refcount is increased, i.e. the steal is on the incref.

All of this seem pretty pointless:
1) we don't need a steal API anymore as we don't pass in a tensor that needs to be stolen
2) (see other comment) we don't actually need two storages here.

Reply:
Yeah, thanks for the background! It would definitely be sweet to get rid of the vestigial degrees of freedom. 

Just to clarify, I was thinking of the `incref(storage)` - figuring that if the storage was literally getting stolen (i.e. destructively acquired, move style) then the abstract decref/incref pair would net out to no change. My guess was that "steal" here really means "alias" or maybe that there was some deeper subtlety, just wanted to drill down. 

...but really, happy not to dedicate neural real estate on an API that's on its way out 😬

Reply:
I believe "steal" here refers to stealing the refcount -- I'm not sure if that fits your definition of "destructively acquired, move style."  It does if we are talking about ref counted pointers, although that's not obvious here.